pkg/restore/restore_crd.go

```diff
@@ -0,0 +1,237 @@
+				_, err := k8sclientset.Kube.CoreV1().ConfigMaps(clusterNamespace).Update(context.TODO(), &cm, v1.UpdateOptions{})
+				if err != nil {
+					scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+				}
+				logging.Info("removed ownerReference for configmap: %s\n", cm.Name)
+			}
+		}
+	}
+
+	services, err := k8sclientset.Kube.CoreV1().Services(clusterNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+	}
+
+	for _, service := range services.Items {
+		for _, owner := range service.OwnerReferences {
+			if string(owner.UID) == targetUID {
+				logging.Info("removing owner references for service %s", service.Name)
+				// Remove ownerReferences.
+				service.OwnerReferences = nil
+
+				// Update the Secret without ownerReferences.
+				_, err := k8sclientset.Kube.CoreV1().Services(clusterNamespace).Update(context.TODO(), &service, v1.UpdateOptions{})
+				if err != nil {
+					scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+				}
+				logging.Info("Removed ownerReference for service: %s\n", service.Name)
+			}
+		}
+	}
+
+	deploys, err := k8sclientset.Kube.AppsV1().Deployments(clusterNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+	}
+
+	for _, deploy := range deploys.Items {
+		for _, owner := range deploy.OwnerReferences {
+			if string(owner.UID) == targetUID {
+				logging.Info("removing owner references for deployemt %s", deploy.Name)
+				// Remove ownerReferences.
+				deploy.OwnerReferences = nil
+
+				// Update the Secret without ownerReferences.
+				_, err := k8sclientset.Kube.AppsV1().Deployments(clusterNamespace).Update(context.TODO(), &deploy, v1.UpdateOptions{})
+				if err != nil {
+					scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+				}
+				logging.Info("Removed ownerReference for deployment: %s\n", deploy.Name)
+			}
+		}
+	}
+
+	pvcs, err := k8sclientset.Kube.CoreV1().PersistentVolumeClaims(clusterNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+	}
+
+	for _, pvc := range pvcs.Items {
+		for _, owner := range pvc.OwnerReferences {
+			if string(owner.UID) == targetUID {
+				logging.Info("removing owner references for pvc %s", pvc.Name)
+				// Remove ownerReferences.
+				pvc.OwnerReferences = nil
+
+				// Update the Secret without ownerReferences.
+				_, err := k8sclientset.Kube.CoreV1().PersistentVolumeClaims(clusterNamespace).Update(context.TODO(), &pvc, v1.UpdateOptions{})
+				if err != nil {
+					scaleBackOperatorOnError(ctx, k8sclientset, operatorNamespace, err)
+				}
+				logging.Info("Removed ownerReference for pvc: %s\n", pvc.Name)
+			}
+		}
+	}
+}
+
+func scaleBackOperatorOnError(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace string, errGot error) {
```

Comment:
Do we have helpers in the plugin already for scaling the operator instead of creating this new func?

Reply:
We do have a helper method to scale the deployments but in command, we first scale the rook operator to 0 and scale back the operator in the end once all the operations are done but if an error occurs between scale 0 to scale 1 the rook operator will be down. So, what I'm doing here is, if any error occurs after scaling the rook operator pod 0 and before scaling back to 1, we'll call this new function where we'll first scale the rook operator back and then return the err.